backend/app/services/shared_task.py

```
@@ -0,0 +1,630 @@
+                )
+
+            # Parse user_id and task_id
+            user_id_str, task_id_str = share_data_str.split("#", 1)
+            try:
+                user_id = int(user_id_str)
+                task_id = int(task_id_str)
+            except ValueError:
+                raise HTTPException(
+                    status_code=400,
+                    detail="Invalid share link format"
+                )
```

Ruff (0.14.7)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)